src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumClient.kt

```diff
@@ -212,31 +212,34 @@ class ElectrumClient(
         listen() // This suspends until the coroutines is cancelled or the socket is closed
     }
 
-    suspend fun send(request: ElectrumRequest, replyTo: CompletableDeferred<ElectrumResponse>) {
+    override suspend fun send(request: ElectrumRequest, replyTo: CompletableDeferred<ElectrumResponse>) {
         mailbox.send(Action.SendToServer(Pair(request, replyTo)))
     }
 
     suspend inline fun <reified T : ElectrumResponse> rpcCall(request: ElectrumRequest): T {
         val replyTo = CompletableDeferred<ElectrumResponse>()
         send(request, replyTo)
-        return replyTo.await() as T
+        return when (val res = replyTo.await()) {
+            is ServerError -> error(res)
```

Comment:
Won't it crash the wallet if we throw an exception here? There are a lot of calls that we seem to make without catching exceptions.

Reply:
Yes it would, but this change doesn't introduce a new behavior.